### PR TITLE
Adjustable timeouts for NuGet

### PR DESCRIPTION
**Pull request by pawld**
Timeout adjustable for sending request and stream processing.

### CODE REVIEW

File: src/Paket.Core/Dependencies/NuGet.fs [hunk 1]
```diff
@@ -880,19 +881,34 @@ let private downloadAndExtractPackage(alternativeProjectRoot, root, isLocalOverr
                     ensureDir targetFileName
 
                     use _trackDownload = Profile.startCategory Profile.Category.NuGetDownload
+                    let! tok = Async.CancellationToken
+                    let taskTimeout =
+                        match Environment.GetEnvironmentVariable("PAKET_RESOLVER_TASK_TIMEOUT") with
```
Comment (forki) [inline src/Paket.Core/Dependencies/NuGet.fs]: can we please get some docs for that?

**Comment by pawld** [comment]
We have updated the functionality in the NuGet.fs and added comments to the timeouts.
I hope that these information are enough.

Please let us know if something is missing or if something is unclear.

**Comment by forki** [comment]
can you please document it in the docs? Not sure where it would belong, but if you don't find a better place you may formulate it as a FAQ and put it in `/docs/content/faq.md`

**Comment by pawld** [comment]
We made a short entry in the FAQ and we hope that's enough.

**Comment by matthid** [comment]
I'm not sure, but shouldn't there be a way to make it work by default without the need to force our users to setup every timeout by hand?

**Comment by Vidarls** [comment]
@forki thanks for merging and deploying. Just tested, and after a painstaking 4 minute DL time, restore succeeded.

**Comment by b0urb4k1** [comment]
Could you close the accompanying issue please?
https://github.com/fsprojects/Paket/issues/3384

@matthid 
There is at least one other timeout that is handled the same way.
PAKET_RESOLVER_TASK_TIMEOUT

Reworking this now is not a big issue i guess, at least now we can work with paket again.
What would be your proposal for the issue? If you do not want the download or write to
timeout in general the issue could be fixed by just setting everything to -1 for the default
i would assume.

**Comment by forki** [comment]
there are multiple people having issues. I'm reverting for now. Let's try to find a solution

**Pull request merged**